timegate/config.py

```diff
@@ -41,33 +43,60 @@ def from_inifile(self, filename, silent=True):
             self['API_TIME_OUT'] = conf.getfloat('server', 'api_time_out')
 
         # Handler configuration
-        if conf.has_option('handler', 'handler_class'):
-            self['HANDLER_MODULE'] = conf.get('handler', 'handler_class')
-        if conf.has_option('handler', 'base_uri'):
-            self['BASE_URI'] = conf.get('handler', 'base_uri')
-        if conf.getboolean('handler', 'is_vcs'):
-            self['RESOURCE_TYPE'] = 'vcs'
-        else:
-            self['RESOURCE_TYPE'] = 'snapshot'
-
-        if conf.has_option('handler', 'use_timemap'):
-            self['USE_TIMEMAPS'] = conf.getboolean('handler', 'use_timemap')
-        else:
-            self['USE_TIMEMAPS'] = False
+        def build_handler(section):
+            """Build handler configuration."""
+            output = {}
+            if conf.has_option(section, 'handler_class'):
+                output['HANDLER_MODULE'] = conf.get(section, 'handler_class')
+            if conf.has_option(section, 'base_uri'):
+                output['BASE_URI'] = conf.get(section, 'base_uri')
+            if conf.getboolean(section, 'is_vcs'):
+                output['RESOURCE_TYPE'] = 'vcs'
+            else:
+                output['RESOURCE_TYPE'] = 'snapshot'
+
+            if conf.has_option(section, 'use_timemap'):
+                output['USE_TIMEMAPS'] = conf.getboolean(section,
+                                                         'use_timemap')
+            else:
+                output['USE_TIMEMAPS'] = False
+            return output
+
+        self.setdefault('HANDLERS', {})
+        for section_name in conf.sections():
+            if section_name == 'handler' or \
+                    section_name.startswith('handler:'):
+                section = build_handler(section_name)
+                if ':' in section_name:
+                    self['HANDLERS'][section_name[len('handler:'):]] = section
+                else:
+                    self.update(section)
```

Comment:
What if we organize the configs of both types of handlers (named and anonymous) on the same level (i.e., under `self['HANDLERS']`)? One option is to use a hard coded placeholder handler name for the anonymous handler such as `DEFAULT` or `ANONYMOUS`, but this will add special cases in the `application.py` file and will limit the use of these reserved keywords as handler names. However, there is a cleaner option to use `None` (not the string `"None"`) as the handler name that will remove special case handling while still organizing all the handlers at the same depth. Here is one possible way to write this (note that I have also added code to deal with the case where a section is names as `handler:` i.e., ending with a colon, but nothing after that.):

```python
        self.setdefault('HANDLERS', {})
        for section_name in conf.sections():
            parts = section_name.rstrip(':').split(':', 1)
            if parts[0] == 'handler':
                section = build_handler(section_name)
                handler_name = None
                if len(parts) > 1:
                    handler_name = parts[1]
                self['HANDLERS'][handler_name] = section
```

This will also remove the following line from `application.py` file:
```python
        self.register_handler(None, self.config)
```